### PR TITLE
Fix #7013: Check PCP for all class references

### LINKED ISSUE #7013: Quote local classes crashes compiler
## minimized code

```Scala
import quoted._

def foo() given QuoteContext = {
  class C
  '[C]
}
```

<details>
<summary>Stack trace</summary>

```scala
java.lang.AssertionError: assertion failed: unresolved symbols: class C(line 10) when pickling examples/i6263.scala while compiling examples/i6263.scala
Exception in thread "main" java.lang.AssertionError: assertion failed: unresolved symbols: class C(line 10) when pickling examples/i6263.scala
	at dotty.DottyPredef$.assertFail(DottyPredef.scala:16)
	at dotty.tools.dotc.core.tasty.TreePickler.pickle(TreePickler.scala:695)
	at dotty.tools.dotc.core.quoted.PickledQuotes$.pickle(PickledQuotes.scala:90)
	at dotty.tools.dotc.core.quoted.PickledQuotes$.pickleQuote(PickledQuotes.scala:32)
	at dotty.tools.dotc.transform.ReifyQuotes$QuoteReifier.pickleAsTasty$1(ReifyQuotes.scala:237)
	at dotty.tools.dotc.transform.ReifyQuotes$QuoteReifier.pickledQuote(ReifyQuotes.scala:245)
	at dotty.tools.dotc.transform.ReifyQuotes$QuoteReifier.transformQuotation(ReifyQuotes.scala:195)
	at dotty.tools.dotc.transform.TreeMapWithStages.transform(TreeMapWithStages.scala:94)
	at dotty.tools.dotc.transform.ReifyQuotes$QuoteReifier.transform(ReifyQuotes.scala:409)
	at dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1269)
	at dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:118)
	at dotty.tools.dotc.transform.TreeMapWithStages.mapOverTree$1(TreeMapWithStages.scala:77)
	at dotty.tools.dotc.transform.TreeMapWithStages.transform(TreeMapWithStages.scala:127)
	at dotty.tools.dotc.transform.ReifyQuotes$QuoteReifier.transform(ReifyQuotes.scala:409)
	at dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1275)
	at dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:90)
	at dotty.tools.dotc.transform.TreeMapWithStages.mapOverTree$1(TreeMapWithStages.scala:77)
	at dotty.tools.dotc.transform.TreeMapWithStages.transform(TreeMapWithStages.scala:106)
	at dotty.tools.dotc.transform.ReifyQuotes$QuoteReifier.transform(ReifyQuotes.scala:409)
	at dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:98)
	at dotty.tools.dotc.transform.TreeMapWithStages.mapOverTree$1(TreeMapWithStages.scala:77)
	at dotty.tools.dotc.transform.TreeMapWithStages.transform(TreeMapWithStages.scala:127)
	at dotty.tools.dotc.transform.ReifyQuotes$QuoteReifier.transform(ReifyQuotes.scala:409)
	at dotty.tools.dotc.ast.TreeMapWithImplicits.traverse$1(TreeMapWithImplicits.scala:52)
	at dotty.tools.dotc.ast.TreeMapWithImplicits.transformStats(TreeMapWithImplicits.scala:59)
	at dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:109)
	at dotty.tools.dotc.transform.TreeMapWithStages.mapOverTree$1(TreeMapWithStages.scala:77)
	at dotty.tools.dotc.transform.TreeMapWithStages.transform(TreeMapWithStages.scala:127)
	at dotty.tools.dotc.transform.ReifyQuotes$QuoteReifier.transform(ReifyQuotes.scala:409)
	at dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1331)
	at dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:102)
	at dotty.tools.dotc.transform.TreeMapWithStages.mapOverTree$1(TreeMapWithStages.scala:77)
	at dotty.tools.dotc.transform.TreeMapWithStages.transform(TreeMapWithStages.scala:127)
	at dotty.tools.dotc.transform.ReifyQuotes$QuoteReifier.transform(ReifyQuotes.scala:409)
	at dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform$$anonfun$2(Trees.scala:1351)
	at scala.collection.immutable.List.mapConserve(List.scala:179)
	at dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1351)
	at dotty.tools.dotc.ast.Trees$Instance$TreeMap.transformStats(Trees.scala:1349)
	at dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1337)
	at dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:102)
	at dotty.tools.dotc.transform.TreeMapWithStages.mapOverTree$1(TreeMapWithStages.scala:77)
	at dotty.tools.dotc.transform.TreeMapWithStages.transform(TreeMapWithStages.scala:127)
	at dotty.tools.dotc.transform.ReifyQuotes$QuoteReifier.transform(ReifyQuotes.scala:409)
	at dotty.tools.dotc.transform.ReifyQuotes$$anon$1.transform(ReifyQuotes.scala:94)
	at dotty.tools.dotc.transform.MacroTransform.run(MacroTransform.scala:21)
	at dotty.tools.dotc.transform.ReifyQuotes.run(ReifyQuotes.scala:90)
	at dotty.tools.dotc.core.Phases$Phase.runOn$$anonfun$1(Phases.scala:316)
	at scala.collection.immutable.List.map(List.scala:286)
	at dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:318)
	at dotty.tools.dotc.Run.runPhases$4$$anonfun$4(Run.scala:158)
	at dotty.runtime.function.JProcedure1.apply(JProcedure1.java:15)
	at dotty.runtime.function.JProcedure1.apply(JProcedure1.java:10)
	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:198)
	at dotty.tools.dotc.Run.runPhases$5(Run.scala:170)
	at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:178)
	at dotty.runtime.function.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:65)
	at dotty.tools.dotc.Run.compileUnits(Run.scala:185)
	at dotty.tools.dotc.Run.compileSources(Run.scala:120)
	at dotty.tools.dotc.Run.compile(Run.scala:104)
	at dotty.tools.dotc.Driver.doCompile(Driver.scala:35)
	at dotty.tools.dotc.Driver.process(Driver.scala:180)
	at dotty.tools.dotc.Driver.process(Driver.scala:149)
	at dotty.tools.dotc.Driver.process(Driver.scala:161)
	at dotty.tools.dotc.Driver.main(Driver.scala:188)
	at dotty.tools.dotc.Main.main(Main.scala)```
</details>

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala [hunk 1]
```diff
@@ -138,8 +138,12 @@ class PCPCheckAndHeal(@constructorOnly ictx: Context) extends TreeMapWithStages(
     /** Is a reference to a class but not `this.type` */
     def isClassRef = sym.isClass && !tp.isInstanceOf[ThisType]
 
-    if (sym.exists && !sym.isStaticOwner && !isClassRef && !levelOK(sym))
+    if (!sym.exists || levelOK(sym))
+      None
+    else if (!sym.isStaticOwner && !isClassRef)
       tryHeal(sym, tp, pos)
+    else if (!sym.owner.isStaticOwner) // local class reference that is phase
+      levelError(sym, tp, pos, "")
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala]: Will a member class be accepted or not? Maybe add a test for that.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala]: They will not. I added a regression test for it.

Comment (liufengyun) [reply compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala]: Maybe update the comment too --- it's not only about local classes.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala]: Done

**Pull request merged**